### DISCUSSION TITLE
Nvim plugin users having to compile Rust every time they PackerSync, how bad could it be?

### DISCUSSION THREAD

**Discussion by ziontee113** [noib3/nvim-oxi#90] [Q&A] [answered]
Hi @noib3 , thank you so much for creating nvim-oxi. I'm planning to use this to create nvim plugins soon. I'm still a beginner with the Rust programming language.

One major question I have before starting developing nvim plugins in Rust is about the end-user experience when installing / updating our plugins that are written in Rust. How bad could their experience be when they have to sit and wait for Rust to compile every time they update their plugins (especially if the plugin is constantly changing).

I've never used a nvim plugin written in Rust before so I have no idea if this is even an issue or not. If not, then it's wonderful! If it is, what can we do to help the end-users?

---
The other small question I have: so by using nvim-oxi, do we have full access to all Rust features? (Like multiple threads etc...) 

Thank you very much! :pray:

**Comment by SCJangra** [comment, marked as answer]
End users don't necessarily have to compile your plugin to actually use it. For example, you can setup a github workflow to automatically compile your plugin for different systems whenever a commit is pushed to your main branch. And your can provide an install/update script to download the latest binaries whehever a user updates their plugin.

**Reply by ziontee113** [reply]
Thank you very much for the response ☺️
